### DISCUSSION TITLE
issue: llama3.2 vision second image not being processed

### DISCUSSION THREAD

**Discussion by danielj23** [open-webui/open-webui#8067] [Issues]
# Bug Report

 

## Installation Method

docker 

## Environment
[v0.3.35](https://github.com/open-webui/open-webui/releases/tag/v0.3.35)
 

Windows 10
Firefox 132.0.1 (64-bit)

**Confirmation:**

- [ x] I have read and followed all the instructions provided in the README.md.
- [ x] I am on the latest version of both Open WebUI and Ollama.
- [ ] I have included the browser console logs.
- [ ] I have included the Docker container logs.
- [x ] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

I should be able to continuously send images to vision-enabled models and have it see the attachment and describe the image.

## Actual Behavior:

I start a conversation with a vision-enabled model (llama3.2-vision) and I attach an image and ask the model to describe the image attached.

The model responds correctly and describes the image.

In the same chat, if I attempt to attach another image, the model replies with "There is no image attached to your message." 

## Description

**Bug Summary:**

open-webui is not passing the image attachment to the model to be able to desribe it.

## Reproduction Details

**Steps to Reproduce:**
start a conversation with a vision-enabled model (llama3.2-vision) and I attach an image and ask the model to describe the image attached.

The model responds correctly and describes the image.

In the same chat, if I attempt to attach another image, the model replies with "There is no image attached to your message." 

## Logs and Screenshots
 
**Screenshots/Screen Recordings (if applicable):**
 
![firefox_fQy59fG0SK](https://github.com/user-attachments/assets/3bd0c1ca-9a82-43c1-abb5-c885df1b78da)

**Comment by RamboRogers** [comment]
Second.  My daughter is super sad. We're getting error 500s

**Comment by tjbck** [comment]
<img width="747" alt="image" src="https://github.com/user-attachments/assets/b07b77a5-d408-44d4-a92b-5e6a703a974c">

Seems to be an issue with llama3.2-vision model, I'm unable to reproduce the issue with llava and other external vision models like gpt-4o.

**Comment by tjbck** [comment]
<img width="650" alt="image" src="https://github.com/user-attachments/assets/08e564a5-37da-418e-82dc-8aa37a6ab4b8">

Can confirm the image is being sent to Ollama as intended in the correct format, perhaps we might want to raise an issue on Ollama repo here.

**Comment by davidamacey** [comment]
There is a limitation with llama3.2-vision, it only accepts one image per request.  Same issue using a vLLM deployment of llama-vision model.  vLLM Docker logs provides a specific error stating the same.

Ref: https://github.com/vllm-project/vllm/issues/8826

**Comment by danktankk** [comment]
> There is a limitation with llama3.2-vision, it only accepts one image per request. Same issue using a vLLM deployment of llama-vision model. vLLM Docker logs provides a specific error stating the same.
> 
> Ref: [vllm-project/vllm#8826](https://github.com/vllm-project/vllm/issues/8826)

@davidamacey I read the other thread, but dont understand exactly what this means.  I am having the same issue where the second image is not being processed.  Does this mean to use another image you basically have to reload the model?  Many thanks for any clarification so those of us not super proficient at this can better understand.

**Comment by danielj23** [comment]
> @davidamacey I read the other thread, but dont understand exactly what this means. I am having the same issue where the second image is not being processed. Does this mean to use another image you basically have to reload the model? Many thanks for any clarification so those of us not super proficient at this can better understand.

I assumed he meant trying to attach two images to a single chat block - which actually doesn't work and throws an error for me 

> Ollama: 500, message='Internal Server Error',

In my opinion, his reference and my issue are separate.   What [tjbck](https://github.com/tjbck) said makes sense and is likely not this app's fault.

**Comment by shabadan** [comment]
Same here! the base64 encoded image is sent in the second request but the reply is odd

![image](https://github.com/user-attachments/assets/df48d093-eda8-474d-ae99-bbae673fc7c8)

**Comment by danktankk** [comment]
> I assumed he meant trying to attach two images to a single chat block - which actually doesn't work and throws an error for me

I add one image and I get whatever information about the image.  Then I add another image at any point later and that image gives me an error every time.  I am not adding more than one image at a time.  It's a little confusing as to what is being said regarding this so I wanted to make that as clear as possible.  If this was already understood, forgive my ignorance, but this doesn't seem like it should be erroring that way I am going about it.  The only way for me to get the next image to process is to restart the service.

**Comment by 925-Studio** [comment]
Same here.

![Screenshot from 2024-12-24 14-49-25](https://github.com/user-attachments/assets/576cbf3d-c580-4c8f-a19f-ded92fd01993)
![Screenshot from 2024-12-24 14-49-36](https://github.com/user-attachments/assets/9e84b519-9f0d-4da2-897c-2ed6912c0f97)
